src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -2352,14 +2352,8 @@ private uint getClassGClayout(CORINFO_CLASS_STRUCT_* cls, byte* gcPtrs)
             uint result = 0;
 
             MetadataType type = (MetadataType)HandleToObject(cls);
-
-            int pointerSize = PointerSize;
-
-            int ptrsCount = AlignmentHelper.AlignUp(type.InstanceFieldSize.AsInt, pointerSize) / pointerSize;
-
-            // Assume no GC pointers at first
-            for (int i = 0; i < ptrsCount; i++)
-                gcPtrs[i] = (byte)CorInfoGCType.TYPE_GC_NONE;
+            uint size = type.IsValueType ? getClassSize(cls) : getHeapClassSize(cls);
```

Comment:
`getHeapClassSize` has `Debug.Assert(_compilation.IsInheritanceChainLayoutFixedInCurrentVersionBubble(type));` assert. What guarantees that this assert is satisfied?

The stack allocation should be disabled for non-valuetype types that are not fixed in the current version bubble. How is it achieved?

Reply:
@jkotas it's checked via a separate JIT-EE API: [canAllocateOnStack](https://github.com/dotnet/runtime/blob/041375910885f3a30ccf0bb8b751566dae7043f4/src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs#L2163-L2166) so JIT will never call this function if `canAllocateOnStack` returned false.